readthedocs/api/v3/views.py

```diff
@@ -565,3 +620,87 @@ class RemoteOrganizationViewSet(
     filterset_class = RemoteOrganizationFilter
     queryset = RemoteOrganization.objects.all()
     permission_classes = (IsAuthenticated,)
+
+
+class UsersViewSet(
+    APIv3Settings,
+    GenericViewSet,
+):
+    # NOTE: this viewset is only useful for nested URLs required for notifications:
+    # /api/v3/users/<username>/notifications/
+    # However, accessing to /api/v3/users/ or /api/v3/users/<username>/ will return 404.
+    # We can implement these endpoints when we need them, tho.
+
+    model = User
+    serializer_class = UserSerializer
+    queryset = User.objects.none()
+    permission_classes = (IsAuthenticated,)
+
+
+class NotificationsUserViewSet(
+    APIv3Settings,
+    ListModelMixin,
+    RetrieveModelMixin,
+    UpdateMixin,
+    UpdateModelMixin,
+    GenericViewSet,
+):
+    model = Notification
+    lookup_field = "pk"
+    lookup_url_kwarg = "notification_pk"
+    serializer_class = NotificationSerializer
+    queryset = Notification.objects.all()
+    permission_classes = (IsAuthenticated,)
+    filterset_class = NotificationFilter
+
+    def get_queryset(self):
+        queryset = super().get_queryset()
+        content_type = ContentType.objects.get_for_model(User)
+        return queryset.filter(
+            attached_to_content_type=content_type,
+            attached_to_id=self.request.user.pk,
+        )
+
+
+class OrganizationsViewSet(
+    APIv3Settings,
+    GenericViewSet,
+):
+    # NOTE: this viewset is only useful for nested URLs required for notifications:
+    # /api/v3/organizations/<slug>/notifications/
+    # However, accessing to /api/v3/organizations/ or /api/v3/organizations/<slug>/ will return 404.
+    # We can implement these endpoints when we need them, tho.
+
+    model = Organization
+    serializer_class = OrganizationSerializer
+    queryset = Organization.objects.none()
+    permission_classes = (IsAuthenticated,)
+
+
+class NotificationsOrganizationViewSet(
+    APIv3Settings,
+    ListModelMixin,
+    RetrieveModelMixin,
+    UpdateMixin,
+    UpdateModelMixin,
+    GenericViewSet,
+):
+    model = Notification
+    lookup_field = "pk"
+    lookup_url_kwarg = "notification_pk"
+    serializer_class = NotificationSerializer
+    queryset = Notification.objects.all()
+    permission_classes = (IsAuthenticated,)
+    filterset_class = NotificationFilter
+
+    def get_queryset(self):
+        queryset = super().get_queryset()
+        content_type = ContentType.objects.get_for_model(Organization)
+        return queryset.filter(
+            attached_to_content_type=content_type,
+            attached_to_id__in=Subquery(
+                AdminPermission.organizations(
+                    self.request.user, admin=True, member=False
+                ).values("id"),
```

Comment:
I never remember which query is the best query for checking admin permission, this could be useful to get someone else's eyes on.

Reply:
`AdminPermission` is _the_ pattern to follow when checking permissions because it supports SSO. If there are still some of the old patterns around, we should replace it with `AdminPermission`.